lib/setopt.c

```diff
@@ -1867,6 +1867,13 @@ CURLcode Curl_vsetopt(struct Curl_easy *data, CURLoption option, va_list param)
      */
     data->set.haproxyprotocol = (0 != va_arg(param, long)) ? TRUE : FALSE;
     break;
+  case CURLOPT_HAPROXY_CLIENT_IP:
+    /*
+     * Set the client IP to send through HAProxy PROXY protocol
+     */
+    result = Curl_setstropt(&data->set.str[STRING_HAPROXY_CLIENT_IP],
+                            va_arg(param, char *));
```

Comment:
Maybe this should set/unset `set.haproxyprotocol`  as well so that it is enough to set/unset this single option to get haproxy with a given IP? Or is there any particular gain to keep them two separate even when this is added?

Reply:
Not really, you are right we should set `haproxyprotocol` with it.

Reply:
Fixed